x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/cluster/routing/allocation/mapper/DataTierFieldMapper.java

```diff
@@ -73,7 +73,21 @@ public Query existsQuery(SearchExecutionContext context) {
 
         @Override
         public ValueFetcher valueFetcher(SearchExecutionContext context, String format) {
-            throw new UnsupportedOperationException("Cannot fetch values for internal field [" + name() + "].");
+            String tierPreference = getTierPreference(context);
+            return lookup -> singletonList(tierPreference);
+        }
+
+        private String getTierPreference(SearchExecutionContext context) {
+            Settings settings = context.getIndexSettings().getSettings();
+            String value = DataTierAllocationDecider.INDEX_ROUTING_PREFER_SETTING.get(context.getIndexSettings().getSettings());
```

Comment:
you save settings in a var above but you don't use it here? 
-> `String value = DataTierAllocationDecider.INDEX_ROUTING_PREFER_SETTING.get(settings);`

Reply:
Oops, thanks for catching that.